code/__DEFINES/is_helpers.dm

```diff
@@ -21,9 +21,12 @@
 
 #define ismutableappearance(D) (istype(D, /mutable_appearance))
 
+#define isweakref(D) (istype(D, /datum/weakref))
+
 #define isimage(D) (istype(D, /image))
 
-#define isweakref(D) (istype(D, /datum/weakref))
+GLOBAL_VAR_INIT(magic_appearance_detecting_image, new /image) // appearances are awful to detect safely, but this seems to be the best way ~ninjanomnom
```

Comment:
this should be reconciled with `IS_APPEARANCE` macro, seen in typeids

alternatively, switch the typeid macros to `IS_TYPEID_*` instead of `IS_*`, since those are implementation defined and definitely not as safe as this (then again; this also seems to be implementation defined, lol)